### ISSUE TITLE
lldb breaking on SIGPIPE on app startup, at runtime

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#10358]
## Steps to Reproduce

From @franklinharper_twitter:

Build Gallery for a physical iOS device (e.g. iPhone 7 Plus) and run in the Xcode debugger. On startup lldb breaks on a couple of `SIGPIPE`s:
```
libsystem_kernel.dylib`writev:
    0x181c609f8 <+0>:  mov    x16, #0x79
    0x181c609fc <+4>:  svc    #0x80
->  0x181c60a00 <+8>:  b.lo   0x181c60a18               ; <+32>
    0x181c60a04 <+12>: stp    x29, x30, [sp, #-0x10]!
    0x181c60a08 <+16>: mov    x29, sp
    0x181c60a0c <+20>: bl     0x181c60f3c               ; cerror
    0x181c60a10 <+24>: mov    sp, x29
    0x181c60a14 <+28>: ldp    x29, x30, [sp], #0x10
    0x181c60a18 <+32>: ret
```

Similarly, switching iOS keyboards (e.g. to Japanese input) and some other events will trigger `SIGPIPE`s.

## Flutter Doctor
```
[✓] Flutter (on Mac OS X 10.12.5 16F73, locale en-AU, channel master)
    • Flutter at /Users/cbracken/src/flutter/flutter
    • Framework revision 4743a806cb (64 minutes ago), 2017-05-26 16:12:42 -0700
    • Engine revision 75c74dc463
    • Tools Dart version 1.24.0-dev.3.0
```

**Comment by cbracken** [comment]
Checking out the backtraces, looks like these come from a few syscalls, `writev` being a pretty good hint:
```
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGPIPE
  * frame #0: 0x0000000180870a00 libsystem_kernel.dylib`writev + 8
    frame #1: 0x0000000180977274 libsystem_trace.dylib`_os_trace_writev + 156
    frame #2: 0x0000000180975a64 libsystem_trace.dylib`_os_log_impl_mirror_to_stderr + 420
    frame #3: 0x00000001809757b8 libsystem_trace.dylib`_os_log_impl_flatten_and_send + 3412
    frame #4: 0x0000000180976934 libsystem_trace.dylib`_os_log_with_args_impl + 388
    frame #5: 0x0000000181876960 CoreFoundation`_CFLogvEx3 + 192
    frame #6: 0x0000000182367450 Foundation`_NSLogv + 128
    frame #7: 0x000000018228e5c8 Foundation`NSLog + 32
    frame #8: 0x0000000187c61058 UIKit`-[UIApplication _checkBackgroundRefreshAPIAdoption] + 324
...
```

I can confirm that eliminating `stdout`, `stderr` forwarding via `asl_log_descriptor` in the engine fixes the issue, though I need to investigate where that leads us with engine logging in general.

Related issues #6758 #8981 (and likely others)

**Comment by cbracken** [comment]
Also related: #9308

**Comment by franklinharper** [comment]
This is now blocking the iOS release of my app.

**Comment by cbracken** [comment]
Fixed in the engine in flutter/engine#3797. Rolled into the framework in #10859.

**Comment by Mishanya840** [comment]
I have the same problem
LOG:
```com.apple.CoreSimulator.SimDevice.DC9781B3-865F-468E-B260-B77FC4AD4DD3[48493] (UIKitApplication:ru.tvip.tv[1f55][rb-legacy][61208]): Service exited due to SIGPIPE | sent by Runner[61208]```

Maybe I need to migrate the project for flutter 1.1.12 or I need to update ios language to swift?

DOCTOR:
``` 
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale ru-RU)
    • Flutter version 1.12.13+hotfix.5 at /Users/mb/development/flutter
    • Framework revision 27321ebbad (6 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/mb/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Users/mb/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/191.6010548/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Android Studio (version 3.5)
    • Android Studio at /Users/mb/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/191.6010548/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.3.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 42.1.4
    • Dart plugin version 193.5731

[✓] IntelliJ IDEA Ultimate Edition (version 2019.3.2 Preview)
    • IntelliJ at /Users/mb/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 42.1.4
    • Dart plugin version 193.5731

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (3 available)
    • MI PLAY           • 195e4f3e0211                             • android-arm64 • Android 8.1.0 (API 27)
    • iPhone (TVIP)     • a00c119f510cf0413442893b1277f860c2ff45d0 • ios           • iOS 12.4.4
    • iPhone 11 Pro Max • DC9781B3-865F-468E-B260-B77FC4AD4DD3     • ios           • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

• No issues found!
 ```
